backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala

```diff
@@ -36,17 +41,55 @@ class VeloxBackend extends Backend {
 
 object VeloxBackendSettings extends BackendSettings {
   override def supportFileFormatRead(format: ReadFileFormat,
-                                     fields: Array[StructField]): Boolean = {
+                                     fields: Array[StructField],
+                                     paths: Seq[String],
+                                     partNonEmpty: Boolean): Boolean = {
+    // Validate partition info.
+    def validatePartInfo: Boolean = {
+      // Fallback to vanilla spark when the input path does not contain the partition info.
+      if (partNonEmpty && !paths.forall(_.contains("="))) {
+        return false
+      }
+      true
+    }
+
+    // Validate compression codec based on file metadata.
+    def validateFileMetaData: Boolean = {
+      val conf = new Configuration()
+      val parquetReadOptions = HadoopReadOptions.builder(conf).build()
+      val supportedCodec = Seq("none", "uncompressed", "snappy", "gzip", "zstd")
+      for (path <- paths) {
+        val hadoopInputFile = HadoopInputFile.fromPath(new Path(path), conf)
+        val parquetFile = ParquetFileReader.open(hadoopInputFile, parquetReadOptions)
+        try {
+          parquetFile.getFooter.getBlocks.forEach(block => {
+            block.getColumns.forEach(column => {
+              if (!supportedCodec.contains(column.getCodec.name().toLowerCase())) {
```

Comment:
This validate may be too exbensive, especially there is many small files which is common in user production environments.
Can we just check the first file or sample to check file? I suppose files with different codec is rarely.

Reply:
Thanks for this suggestion. Fixed.